### PR TITLE
Adds detection of new Mastercard BIN ranges

### PR DESCRIPTION
**Pull request by chrissrogers**
- To be made available Oct 2016

### CODE REVIEW

**Comment by chrissrogers** [comment]
Heads up @isaachall

File: lib/recurly/validate.js [hunk 1]
```diff
@@ -19,7 +19,14 @@ const TYPES = [
   },
   {
     type: 'master',
-    pattern: /^5[0-5]/,
+    pattern: {
+      test: (n) => {
+        let firstSix = parseInt(n.substr(0, 6), 10);
+        if (/^5[0-5]/.test(n)) return true;
```
Comment (isaachall) [inline lib/recurly/validate.js]: This can be tightened just a little... MasterCards do not start with "50", they can start with "5[1-5]".

**Comment by snodgrass23** [comment]
:+1:

**Comment by isaachall** [comment]
Awesome, thanks @chrissrogers !

**Pull request merged**